test/gtest/uct/uct_test.cc

```diff
@@ -183,6 +187,18 @@ std::vector<const resource*> uct_test::enum_resources(const std::string& tl_name
             ASSERT_UCS_OK(status);
 
             for (unsigned j = 0; j < num_tl_resources; ++j) {
+                if ((tl_resources[j].tl_name == static_cast<std::string>("tcp")) &&
+                    (tcp_devs != "all")) {
+                    if (tcp_dev_set) {
+                        continue;
+                    } else {
+                        if ((tcp_devs == "") || (tcp_devs == "fastest") ||
```

Comment:
How does it work? Is `tl_resources` sorted by speed?

Reply:
Yes, TCP provides a list of devvice (IP interfaces) sorted by their speeds. We rely on this fact, added the comment